source/vibe/core/core.d

```diff
@@ -877,18 +892,64 @@ unittest {
 
 
 /**
-	Returns a new armed timer.
-
-	Note that timers can only work if an event loop is running, explicitly or
-	implicitly by running a blocking operation, such as `sleep` or `File.read`.
+	Creates a new timer, that will fire `callback` after `timeout`
+
+    There are two kinds of timer: one-off timers and periodic timers.
+    One-off timers fire only once, after a specific amount of time,
+    while periodic timer fire at a regular interval.
+
+    One-off_timers:
+    One-off timers can be used for performing a task after a specific delay,
+    or for the initial setup of an application. For example, the following
+    is a common pattern for a Vibe.d application:
+    ---
+    import vibe.d;
+
+    /// Setup the application and let other server know we are running
+    void setup (Config conf) {
+        // Magic!
+    }
+
+    int main (string[] args) {
+        Config conf = parseArguments(&args);
+        setTimer(0.msecs, () => setup(conf));
+        return runApplication(&args);
+    }
+    ---
+
+    As timers only run when the event loop is running, and `runApplication`
+    will start the event loop, `setupTimer` will just register the closure
+    that calls `setup` to be executed as soon as the event loop is ready.
```

Comment:
This is not the full truth, as any of vibe.d's "blocking" functions will run a temporary event loop. So as long as the thread is not stuck with a pure CPU load or `Thread.sleep`, the timer should be invoked at the right time regardless of `runApplication`.